### PR TITLE
Transform f1 xmssn line

### PR DESCRIPTION
**Pull request by aesharpe**
### TLDR

This PR adds the transmission table to the list of transformed FERC1 tables. It is the last remaining non row-literal table, and was relatively strait forward.

I created a new transform class for this table called `replace_with_na` that allows you to replace certain values with NA without needed to categorize the rest of the values. 

The transforms I applied to the table are:
* replace_with_na
* drop_invalid_rows

#### col: `supporting_structure_type`

I wanted to apply a string categorization on the `supporting_structure_type` column. The definition in the XBRL clearly states that the field should be one of four things: 

> Supporting structure can be: (1) single pole wood or steel; (2) H-frame wood, or steel poles; (3) tower; or (4) underground construction If a transmission line has more than one type of supporting structure.

However, the actual content of the field spans way beyond what can reasonably be inferred as one of those four categories. My options here are: 
a) Categorize all values regardless, nulling out those that are un-categorizable.
b) Map recognizable values and leave unrecognizable values as they are.
c) Leave all values as they are.

Right now, I've done nothing (c), but I would consider doing option (b). I'm not likely to consider option (a) because I'm don't want to drop lots of information without knowing more about what it actually means. Granted, I asked my roomate that works on transmission lines what those other, un-categorizable values could be and they didn't know what they were...

Here's an example of the type of values in `supporting_structure_type`: 
```
'H-FRAME',
 'WOOD POLE',
 'H-Wood',
 'SP-Wood',
 'SP-Steel',
 'Underground',
'HFW & SPS',
 'HFW & ST',
 'HFW & SPW',
 '      T',
 '  W-H Fr.',
 '  DC-CP',
'H-.74',
 'P-.76;H-1.21',
 'Z-.04;H-9.03;',
 'HH-2.45',
 'Z-1.10',
 'P-3.87;H-18.68',
```
#### col: `conductor_size_and_material `

This column is also very messy and not something I could easily parse. As the title suggests, there are two types of information embedded in one column, and I don't know enough about conductor size or material (and the reporting isn't consistent enough) to separate them out or standardize them. For now, this is just a regular string column with all sorts of information in there...

If we wanted, we could do more research on conductor size and material and at least null out some useless values. 

### Notes

I have yet to include this table in the output layer.

### CODE REVIEW

File: src/pudl/transform/ferc1.py [hunk 1]
```diff
@@ -2584,6 +2585,26 @@ def spot_fix_rows(self, df: pd.DataFrame) -> pd.DataFrame:
...
 
 
+class TransmissionFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """A table transformer specific to the :ref:`transmission_ferc1` table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.TRANSMISSION_FERC1
+
+    @cache_df(key="main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Table specific transforms for plants_small_ferc1.
```
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:2595]: small plants?

File: src/pudl/transform/ferc1.py [hunk 2]
```diff
@@ -2584,6 +2585,26 @@ def spot_fix_rows(self, df: pd.DataFrame) -> pd.DataFrame:
...
+
+        Params:
+            df: Pre-processed, concatenated XBRL and DBF data.
+
+        Returns:
+            A single transformed table concatenating multiple years of cleaned data
+            derived from the raw DBF and/or XBRL inputs.
+        """
+        df = self.replace_with_na(df).pipe(self.drop_invalid_rows)
```
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:2604]: is there anything else from the `Ferc1AbstractTableTransformer.transform_main` that you might want for this table? you can always do `super().transform_main(df).pipe(self.replace_with_na)` if you want to tack something onto the end of the standard method. most of the default steps in `transform_main` do nothing if there are no defined params for that step so even if there are some that you don't need that's fine. I think it's good to add things into the default if possible. Like maybe even add this `replace_with_na` into the default `transform_main` which will  do nothing to all of the other tables that don't have a param for it..

**Review by cmgosnell — CHANGES_REQUESTED** [review]
You'll need to add these tables into the settings files (both fast and full). for the fast file, you'll need to add the raw tables into the `ferc_to_sqlite_settings` section as well as the pudl table name in the pudl etl section.

1 small docs suggestion and 1 possible suggestion about moving use the default `transform_main`

File: src/pudl/transform/ferc1.py (hunk 2 above)
Comment (aesharpe) [reply src/pudl/transform/ferc1.py:2604]: Hmm ok I guess I wasn't really doing this with the small plants table because the order of opperations was different (i.e., we needed to clean the columns before categorizing them.)

**Comment by aesharpe** [comment]
It's worth noting that there are some records with duplicate `record_id` values. The other information differs slightly. For example: 

|        | record_id                      |   utility_id_ferc1 |   report_year | start_point     | end_point   |   operating_voltage_kv |   designed_voltage_kv | supporting_structure_type   |   transmission_line_length_miles |   transmission_line_and_structures_length_miles |   num_transmission_circuits | conductor_size_and_material   |   capex_land |   capex_other |   capex_total |   opex_operations |   opex_maintenance |   opex_rents |   opex_total |
|-------:|:-------------------------------|-------------------:|--------------:|:----------------|:------------|-----------------------:|----------------------:|:----------------------------|---------------------------------:|------------------------------------------------:|----------------------------:|:------------------------------|-------------:|--------------:|--------------:|------------------:|-------------------:|-------------:|-------------:|
| 132342 | f1_xmssn_line_2000_12_32_17_14 |                165 |          2000 | 11415 NORTHWEST | DEVON       |                    138 |                   138 | UG                          |                             0.3  |                                               0 |                           1 | 2500 1/C CU                   |          nan |           nan |           nan |               nan |                nan |          nan |          nan |
| 132343 | f1_xmssn_line_2000_12_32_17_14 |                165 |          2000 | 11414 NORTHWEST | DEVON       |                    138 |                   138 | UG                          |                             0.18 |                                               0 |                           1 | 2500 1/C CU                   |          nan |           nan |           nan |               nan |                nan |          nan |          nan |


and 


|        | record_id                                                    |   utility_id_ferc1 |   report_year | start_point   | end_point   |   operating_voltage_kv |   designed_voltage_kv | supporting_structure_type   |   transmission_line_length_miles |   transmission_line_and_structures_length_miles | num_transmission_circuits   | conductor_size_and_material   |      capex_land |   capex_other |   capex_total |   opex_operations |   opex_maintenance |   opex_rents |   opex_total |
|-------:|:-------------------------------------------------------------|-------------------:|--------------:|:--------------|:------------|-----------------------:|----------------------:|:----------------------------|---------------------------------:|------------------------------------------------:|:----------------------------|:------------------------------|----------------:|--------------:|--------------:|------------------:|-------------------:|-------------:|-------------:|
| 563551 | transmission_line_statistics_422_2021_c000317_345_kv_lines__ |                178 |          2021 | 345 KV Lines  | <NA>        |                    nan |                   nan | <NA>                        |                           nan    |                                             nan | <NA>                        | <NA>                          |     1.16458e+06 | nan           |   1.16458e+06 |               nan |                nan |          nan |          nan |
| 563556 | transmission_line_statistics_422_2021_c000317_345_kv_lines__ |                178 |          2021 | 34.5 KV Lines | <NA>        |                    nan |                   nan | <NA>                        |                             7.31 |                                             nan | <NA>                        | <NA>                          | 56036           |   3.37056e+06 |   3.4266e+06  |               nan |                nan |          nan |          nan |

**Comment by cmgosnell** [comment]
I think this is failing bc you haven't added the raw tables into the fast settings file. see my previous comment.

also on the non-unique record_id, how many dupes are there overall? and do they look like truly different data (which these two examples do look actually different to me at least). overall my main question is whether we actually believe this table should have unique record ids. I would start by looking at these records raw tables to just double check to see whether or not there is something weird in the raw records or whether we have transformed them weirdly.

**Comment by aesharpe** [comment]
> how many dupes are there overall?

There are 8 DBF dups and 12 XBRL dups

**Comment by aesharpe** [comment]
> my main question is whether we actually believe this table should have unique record ids. I would start by looking at these records raw tables to just double check to see whether or not there is something weird in the raw records or whether we have transformed them weirdly.

Right now, I think it makes sense to create record ids.

**Comment by aesharpe** [comment]
I looked into whether there was a primary key for this table, and I'm not convinced that there is. The obvious `report_year, utility_id_ferc1, start_point, end_point` did not yield a unique dataset. It seems like the kind of thing that, while unlikely, it is possible that a utility could have two identical transmission lines going from point A to point B meaning that even full row duplicates should probably be entertained.

File: src/pudl/transform/ferc1.py [hunk 3]
```diff
@@ -2607,6 +2609,26 @@ def spot_fix_rows(self, df: pd.DataFrame) -> pd.DataFrame:
...
+        """Table specific transforms for transmission_ferc1.
+
+        Params:
+            df: Pre-processed, concatenated XBRL and DBF data.
+
+        Returns:
+            A single transformed table concatenating multiple years of cleaned data
+            derived from the raw DBF and/or XBRL inputs.
+        """
+        return super().transform_main(df)
```
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:2629]: since you are only running the `super` version of this method, you can delete this whole table-specific method

File: src/pudl/metadata/fields.py [hunk 4]
```diff
@@ -1999,6 +2042,23 @@
...
+        },
+        "capex_total": {"description": "Total costs for the transmission line."},
+        "opex_operations": {
+            "description": "Operating expenses for the transmission line."
+        },
+        "opex_maintenance": {
+            "description": "Maintenance expenses for the transmission line."
+        },
+        "opex_rents": {"description": "Rent expenses for the transmission line."},
+        "opex_total": {"description": "Overall expenses for the transmission line."},
```
Comment (cmgosnell) [inline src/pudl/metadata/fields.py:2060]: can you add types for these?

**Review by cmgosnell — CHANGES_REQUESTED** [review]
hey! can you delete the notebook (we talked about this and noted that it wasn't supposed to be added into this pr) plus bb changes below. I'll poke back in here quickly once you address these to get this merged iiiiiin

File: src/pudl/metadata/fields.py (hunk 4 above)
Comment (aesharpe) [reply src/pudl/metadata/fields.py:2060]: I wasn't sure if these needed types because they are already given types in the `FIELD_METADATA` dict. Does this `FIELD_METADATA_BY_RESOURCE` fully override whatever is in `FIELD_METADATA` or does it supplement / replace whatever is in there?

Comment (cmgosnell) [reply src/pudl/metadata/fields.py:2060]: oh your very right my bad!
